### ISSUE TITLE
The noteheads do not change their shape after edit staff type

### ISSUE THREAD

**Issue by Dima-S-Jr** [musescore/MuseScore#22733]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Editing the noteheads in the edit of the drumset does not affect their final shape in the score.

### Steps to reproduce

1. Create a glockenspiel staff.
2. Add a edit staff type anywhere by changing the instrument in parallel.
3. Open the edit drumdset, change the shape of the notehead to any other.
4. The shape of the notehead of the score will not change.

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/134041121/e778dcf1-4d1a-4c3d-aa81-7efb52880766

### MuseScore Version

OS: Windows 10 Version 2009MuseScore Studio version (64-bit): 4.3.0-241231433, revision: github-musescore-musescore-5f36e74

### Regression

I don't know

### Operating system

Windows 11

### Additional context

I noticed that this happens only when a "custom" notehead group is assigned in the edit drumset.
![Снимок экрана (322)](https://github.com/musescore/MuseScore/assets/134041121/26193815-4ec2-4596-accb-3698e511d7bf)

**Comment by Dima-S-Jr** [comment]
I also noticed that a score with a custom group of noteheads cannot be uploaded to musescore.com . A few hours after uploading the score, page 404 "Page not found" will appear. 
@bkunda, should I publish this as a separate issue, or will my comment be enough to solve this together with the current issue?

**Comment by zacjansheski** [comment]
Doing this in earlier versions of MuseScore 4 actually crashes the program, so technically not a regression...
